src/stdlib/Json.ts

```diff
@@ -0,0 +1,127 @@
+import { AssociativeArray } from "../brsTypes/components/AssociativeArray";
+import { BrsArray } from "../brsTypes/components/BrsArray";
+import { Interpreter } from "../interpreter";
+import { Literal } from "../parser/Expression";
+import { randomBytes } from "crypto";
+
+import {
+    BrsBoolean,
+    BrsInvalid,
+    BrsString,
+    BrsType,
+    Callable,
+    Float,
+    Int32,
+    Int64,
+    ValueKind
+} from "../brsTypes";
+
+function brsValueOf(x: any): any {
+    if (x === null) { return BrsInvalid.Instance; }
+    let t: string = typeof x;
+    switch (t) {
+    case "boolean":
+        return BrsBoolean.from(x);
+    case "string":
+        return new BrsString(x);
+    case "number":
+        if (Number.isInteger(x)) {
+            return x >= -2_147_483_648 && x <= 2_147_483_647 ? new Int32(x) : new Int64(x);
+        }
+        return new Float(x);
+    case "object":
+        if (Array.isArray(x)) {
+            return new BrsArray(x.map(brsValueOf));
+        }
+        return new AssociativeArray(
+            Object.getOwnPropertyNames(x).map((k: string) => {
+                return {
+                    name: brsValueOf(k),
+                    value: brsValueOf(x[k])
+                };
+            })
+        );
+    default:
+        throw new Error(`brsValueOf not implemented for: ${x} <${t}>`);
+    }
+}
+
+function jsonOf(interpreter: Interpreter, x: BrsType, uid: BrsString): string {
```

Comment:
Please add a JSDoc comment to `jsonOf` and `brsValueOf` to describe what they're used for and what significance the parameters and return values have.  I know they're not exported, but they've got some pretty significant chunks of logic and the meanings of everything aren't immediately obvious.

Reply:
done